torchgeo/datasets/cloud_cover.py

```diff
@@ -0,0 +1,389 @@
+            if link["rel"] == "item"
+        ]
+        label_collection_item_hrefs = sorted(label_collection_item_hrefs)
+
+        for label_href in label_collection_item_hrefs:
+            label_json = os.path.join(label_collection_path, label_href)
+            indexed_item = self._load_items(label_json)
+            indexed_chips.append(indexed_item)
+
+        return indexed_chips
+
+    def _validate_bands(self, bands: Sequence[str]) -> None:
+        """Validate list of bands.
+
+        Args:
+            bands: user-provided tuple of bands to load
+
+        Raises:
+            AssertionError: if ``bands`` is not a tuple
+            ValueError: if an invalid band name is provided
+        """
+        for band in bands:
+            if band not in self.band_names:
+                raise ValueError(f"'{band}' is an invalid band name.")
+
+    def _check_integrity(self) -> bool:
+        """Check integrity of dataset.
+
+        Returns:
+            True if dataset files are found and/or MD5s match, else False
+        """
+        images: bool = check_integrity(
+            os.path.join(self.root, self.image_meta[self.split]["filename"]),
+            self.image_meta[self.split]["md5"] if self.checksum else None,
+        )
+
+        targets: bool = check_integrity(
+            os.path.join(self.root, self.target_meta[self.split]["filename"]),
+            self.target_meta[self.split]["md5"] if self.checksum else None,
+        )
+
+        return images and targets
+
+    def _download(self, api_key: Optional[str] = None) -> None:
+        """Download the dataset and extract it.
+
+        Args:
+            api_key: a RadiantEarth MLHub API key to use for downloading the dataset
+
+        Raises:
+            RuntimeError: if download doesn't work correctly or checksums don't match
+        """
+        if self._check_integrity():
+            print("Files already downloaded and verified")
+            return
+
+        download_radiant_mlhub_dataset(self.dataset_id, self.root, api_key)
+
+        image_archive_path = os.path.join(
+            self.root, self.image_meta[self.split]["filename"]
+        )
+        target_archive_path = os.path.join(
+            self.root, self.target_meta[self.split]["filename"]
+        )
+        for fn in [image_archive_path, target_archive_path]:
+            extract_archive(fn, self.root)
+
+    def plot(
+        self,
+        sample: Dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            time_step: time step at which to access image, beginning with 0
+            suptitle: optional suptitle to use for figure
+
+        Returns:
+            a matplotlib Figure with the rendered sample
```

Comment:
Should have a "Raises" section

Reply:
What would this likely raise? For the most part the plot method was copied using some of the other datasets as a template.

Reply:
It raises ValueError a few lines later